### PR TITLE
Migration from Swift 2.3 to Swift 3.0

### PR DESCRIPTION
**Pull request by mitulmanish**

### CODE REVIEW

**Comment by tatey** [comment]
Thank you! I will try to review shortly. Initial impression is :+1:

File: Source/Client.swift [hunk 1]
```diff
@@ -9,128 +9,129 @@ public class Client {
...
-
+	
 	private var observers: [ClientObserver]
-
+	
 	public convenience init(accessToken: String, lights: [Light]? = nil, scenes: [Scene]? = nil) {
 		self.init(session: HTTPSession(accessToken: accessToken), lights: lights, scenes: scenes)
 	}
-
+
```
Comment (tatey) [inline Source/Client.swift:17]: Please delete this extra whitespace. I know Xcode sucks at adding whitespace, but I try to catch it in diffs.

File: Source/ClientObserver.swift [hunk 2]
```diff
@@ -6,11 +6,11 @@
...
 
 class ClientObserver {
-	typealias LightsDidUpdate = (lights: [Light]) -> Void
-
-	let lightsDidUpdateHandler: LightsDidUpdate
-
-	init(lightsDidUpdateHandler: LightsDidUpdate) {
+	typealias LightsDidUpdate = (_ lights: [Light]) -> Void
+	
+	let lightsDidUpdateHandler: LightsDidUpdate?
```
Comment (tatey) [inline Source/ClientObserver.swift]: What's the motivation for making this property optional? The original intent behind this class was to let you bind observers to listen for changes to the client. If you intend to have a nil callback, why not just skip setting up an observer?

Would you be able to share a use case?

File: Source/Light.swift [hunk 3]
```diff
@@ -10,24 +10,40 @@ public struct Light: Equatable, CustomStringConvertible {
 	public let power: Bool
 	public let brightness: Double
 	public let color: Color
+	public let productInformation: ProductInformation?
```
Comment (tatey) [inline Source/Light.swift]: I would have preferred to see this called `productInfo` since that would feel closer to Cocoa (Eg: `userInfo`).

**Review by tatey — CHANGES_REQUESTED** [review]
Hey @mitulmanish. Thanks a tonne for making this PR. I like the addition of `ProductInformation` and all the Swift 3 goodness.

Before merging there's a couple of things I'd like help with:

* Curious to understand more about making the handler properties on `ClientObserver` and `LightObserver` optional.
* There's a tonne of additional whitespace that's resulting in a lot more changes than needed. Would you mind cleaning this up?

Otherwise I'd love to merge this in. Please feel free to ask me anything. I'm also happy to help.

**Comment by mitulmanish** [comment]
@tatey Thanks for the detailed feedback.I have made the necessary changes.The changes currently live on https://github.com/LIFX/LIFXHTTPKit/pull/3.I will be testing this branch locally and if all is good I will push the changes upstream.

**Comment by tatey** [comment]
Great. Look forward to it.

**Review by tatey — APPROVED** [review]

**Comment by tatey** [comment]
Thanks a tonne for the patch.

**Comment by tatey** [comment]
I've pushed out a new release https://github.com/tatey/LIFXHTTPKit/releases/tag/3.0.0.

**Pull request merged**